core/src/consensus/tendermint/types.rs

```diff
@@ -63,3 +68,87 @@ impl Encodable for Step {
         s.append_single_value(&self.number());
     }
 }
+
+const MAX_VALIDATOR_SIZE: usize = 800;
+const BITSET_SIZE: usize = MAX_VALIDATOR_SIZE / 8;
+
+#[derive(Copy, Clone)]
+pub struct BitSet([u8; BITSET_SIZE]);
+
+impl fmt::Debug for BitSet {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        self.0[..].fmt(formatter)
+    }
+}
+
+impl BitSet {
+    pub fn new() -> Self {
+        BitSet([0; BITSET_SIZE])
+    }
+
+    pub fn from_bytes(bytes: Bytes) -> Self {
+        let mut bit_set = BitSet::new();
+        for (i, byte) in bytes.iter().enumerate() {
+            bit_set.0[i] = *byte;
+        }
+        bit_set
+    }
+
+    pub fn all_set() -> Self {
+        let mut bit_set = BitSet::new();
+        for i in 0..bit_set.0.len() {
+            bit_set.0[i] = u8::max_value()
+        }
+        bit_set
+    }
+
+    pub fn into_bytes(self) -> Bytes {
+        self.0.to_vec()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        // An array whose size is bigger tha n 32 does not have PartialEq trait
+        // So coerce it to slice to compare
+        let own: &[u8] = &self.0;
+        let empty: &[u8] = &[0; BITSET_SIZE];
+        own == empty
+    }
+
+    pub fn set(&mut self, index: usize) {
```

Comment:
`IndexMut` should return a mutable reference of boolean value.
Since we save the 8 bits as `u8` type, we could not return a reference of boolean.